framework/docker/chain_node.go

```diff
@@ -42,30 +44,31 @@ const (
 	privValPort = "1234/tcp"
 )
 
-func (n *ChainNode) GetInternalPeerAddress(ctx context.Context) (string, error) {
-	id, err := n.nodeID(ctx)
+// GetInternalPeerAddress retrieves the internal peer address for the ChainNode using its ID, hostname, and default port.
+func (cn *ChainNode) GetInternalPeerAddress(ctx context.Context) (string, error) {
+	id, err := cn.nodeID(ctx)
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%s@%s:%d", id, n.HostName(), 26656), nil
+	return fmt.Sprintf("%s@%s:%d", id, cn.HostName(), 26656), nil
 }
 
-func (n *ChainNode) GetInternalRPCAddress(ctx context.Context) (string, error) {
-	id, err := n.nodeID(ctx)
+// GetInternalRPCAddress returns the internal RPC address of the chain node in the format "nodeID@hostname:port".
+func (cn *ChainNode) GetInternalRPCAddress(ctx context.Context) (string, error) {
+	id, err := cn.nodeID(ctx)
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%s@%s:%d", id, n.HostName(), 26657), nil
+	return fmt.Sprintf("%s@%s:%d", id, cn.HostName(), 26657), nil
 }
 
 type ChainNodes []*ChainNode
 
 type ChainNode struct {
+	ChainNodeParams
```

Comment:
all of the ChainNode specific fields were pulled out into this type. It made dealing with things a little cleaner.